### PR TITLE
Registrations support in UI layer

### PR DESCRIPTION
**Pull request by ladislavdubravsky**
The core change happens in `src/name_bazaar/ui/spec.cljs` - here the data structures containing registrar information on TLD labelhashes are specified. From this point we track their usage doing corresponding changes wherever needed.

### CODE REVIEW

**Comment by ladislavdubravsky** [comment]
> So this was the problem all along?

I'm beyond restarting and clicking through everything to make nuanced experiments 15 mins each, but I think your gas fix idea was crucial. Using unintended (and different) version of OZS to what is put as ethregistrar requirement didn't help either I think.

File: run-ganache.sh [hunk 1]
```diff
@@ -3,6 +3,6 @@
 docker stop namebazaar-ganache || true
 docker rm namebazaar-ganache || true
 
-docker run --name=namebazaar-ganache -d \
+docker run --name=namebazaar-ganache \
```
Comment (Siegrift) [inline run-ganache.sh:6]: Nice. Although I hardly believe anything meaningful will come out of those logs :/

File: src/name_bazaar/server/contracts_api/registrar.cljs [hunk 2]
```diff
@@ -31,7 +31,7 @@
                            opts))})
 
 (defn registration-owner [{:keys [:ens.record/hash :ens.record/label]}]
-  (contract-call :name-bazaar-registrar :ownerOf (sha3 label hash)))
+  (contract-call :name-bazaar-registrar :owner-of (sha3 label hash)))
```
Comment (Siegrift) [inline src/name_bazaar/server/contracts_api/registrar.cljs:34]: Also, I checked deep down in the district smart contracts dependency and both camelCase version and kebab-case should work. But we use kebab case for consistency.

File: src/name_bazaar/ui/events.cljs [hunk 3]
```diff
@@ -88,14 +88,14 @@
...
 
-    :route.registrar/register
-    (cond
-      (not (nil? (:name query-params)))
-      {:dispatch-n [[:registration-bids.states/load]
-                    [:registration-bids.state/load (sha3 (-> query-params
-                                                             :name
-                                                             nb-ui-utils/strip-root-registrar-suffix))]]}
-      :else {:dispatch [:registration-bids.states/load]})
+    ;:route.registrar/register
```
Comment (Siegrift) [inline src/name_bazaar/ui/events.cljs]: Add a comment why this is commented so we know why in the future.

File: src/name_bazaar/ui/events/registrar_events.cljs [hunk 4]
```diff
@@ -51,166 +52,206 @@
...
-                          (gstring/format "Bid for %s successfully revealed"
-                                          ens-record-name)]]
-
-                        (contains? #{:new-bid :start-auctions-and-bid} method)
-                        [(if (= :start-auctions-and-bid method)
-                           [:district0x.snackbar/show-message
-                            (gstring/format " %s registration successfully opened and bid placed" ens-record-name)]
-                           [:district0x.snackbar/show-message
-                            (gstring/format "Bid successfully placed for %s" ens-record-name)])])}))
+;(defn- registrar-transaction [method {:keys [:name-bazaar-registrar/label :name-bazaar-registrar/bidder :name-bazaar-registrar/bid-salt
```
Comment (Siegrift) [inline src/name_bazaar/ui/events/registrar_events.cljs]: comment

File: src/name_bazaar/ui/events/registrar_events.cljs [hunk 5]
```diff
@@ -51,166 +52,206 @@
...
 (reg-event-fx
-  :name-bazaar-registrar-entry.deed.owner/loaded
+  :name-bazaar-registrar.registration/loaded
   interceptors
-  (fn [{:keys [:db]} [label-hash deed-owner]]
-    {:db (assoc-in db [:name-bazaar-registrar/entries label-hash :name-bazaar-registrar.entry.deed/owner] deed-owner)}))
+  (fn [{:keys [:db]} [label-hash db-keyword value]]
+    {:db (assoc-in db [:name-bazaar-registrar/registrations label-hash db-keyword] value)}))
+
+;(reg-event-fx
```
Comment (Siegrift) [inline src/name_bazaar/ui/events/registrar_events.cljs]: comment

File: src/name_bazaar/ui/pages/register_name_page.cljs [hunk 6]
```diff
@@ -15,520 +15,521 @@
...
-      (add-watch label :label-watcher (watch-fn))
-      (if (and name (nb-ui-utils/valid-ens-name? name))
-        (reset! label (nb-ui-utils/strip-root-registrar-suffix (nb-ui-utils/normalize name)))
-        (reset! label ""))
-      [app-layout/app-layout {:meta {:title "NameBazaar - Register ENS Name"
-                                     :description "Simplest way to register a new ENS name."}}
-       [ui/Segment
-        [:h1.ui.header.padded "Register Name"]
-        [register-name-form]]])))
+;
```
Comment (Siegrift) [inline src/name_bazaar/ui/pages/register_name_page.cljs]: comment

File: src/name_bazaar/ui/subs/registrar_subs.cljs [hunk 7]
```diff
@@ -5,94 +5,94 @@
...
 
-(defn query-end-bidding-date
-  [registration-date {:keys [:minutes :hours] :as reveal-period}]
-  (t/minus registration-date
-           (cond
-             minutes
-             (t/minutes minutes)
-             hours
-             (t/hours hours))))
+;(defn query-end-bidding-date
```
Comment (Siegrift) [inline src/name_bazaar/ui/subs/registrar_subs.cljs]: comment

**Review by Siegrift — APPROVED** [review]
👍, but please add comments (TODOs) why some stuff is commented out. 
I'll try to write some tests for this when it's merged.

Btw. there are still some updates to do right? We are still not using the latest ENS contracts, right?

**Comment by ladislavdubravsky** [comment]
> Btw. there are still some updates to do right? We are still not using the latest ENS contracts, right?

We were.
But there was a fresh update in January, e.g. finally removing the deprecated contracts which were there for no good reason and confused me earlier.
But I don't think we're jumping on that. Another thing that happened is Solidity update to 0.7, and at the same time this didn't happen in ethregistrar repo yet and I don't think it'll be compilable with 0.7.

**Review by Siegrift — APPROVED** [review]

**Pull request merged**